frontend/src/metabase/dashboard/components/DashboardHeaderView.styled.tsx

```diff
@@ -34,6 +37,23 @@ export const HeaderRow = styled(
   ${breakpointMaxSmall} {
     flex-direction: column;
     align-items: baseline;
+  }
+`;
+
+export const HeaderContainer = styled.div<{ isSidebarOpen: boolean }>`
+  ${props =>
+    props.isSidebarOpen &&
+    css`
+      margin-right: ${SIDEBAR_WIDTH}px;
+    `}
+
+  ${breakpointMaxMedium} {
+    margin-right: 0;
+  }
+`;
```

Comment:
Adjust dashboard header position to align with dashboard body positioning when dashboard sidebar is open. See comment here for reference: https://github.com/metabase/metabase/pull/38634/files#diff-382948855bb5f3cf36bc9f62e5b266b4d7943473b4b30b4fd26890062f5eddbbR120